hazelcast/include/hazelcast/client/IMap.h

```diff
@@ -479,7 +479,7 @@ namespace hazelcast {
                                         getName(), getContext().getClientClusterService(),
                                         getContext().getSerializationService(),
                                         std::move(listener),
-                                        includeValue, getContext().getLogger())), includeValue);
+                                        includeValue, getContext().getLogger())), includeValue, listener.flags);
```

Comment:
parameter order may be a problem. you are accessing flags field of the moved object !!!

Reply:
fixed